### DISCUSSION TITLE
[Task]: Configuration sharing (like `_BASE_` in detectron2) + kwargs overwriting

### DISCUSSION THREAD

**Discussion by DominikFilipiak** [open-edge-platform/anomalib#1250] [Feature Requests]
### What is the motivation for this task?

1. When doing experiments, we often make small changes to the YAML config by hand and see what's the output of the model. 
These configs will share the vast majority of their parameters.
Since the number of configs quickly rises up, it's easy to get lost. 
Therefore, I think it would be nice to have a config inheritance mechanism. 
So, we'd have a `base config`, and a set of additional configs. 
We pass `additional config` to the train/test script and pull everything from `base` (location defined by, say, `_BASE_` keyword in `additional config`) and overwrite keys defined in `additional`. 
This forms a `combined` config, which is the final one.

2. Additionally, if a user will provide a keyword argument (e.g. `--ops model.n_features=50 model.backbone=resnet18`) to the script, it should overwrite values in the `combined` config. 
I think it should be implemented in train/test scripts

### Describe the solution you'd like

It's nicely done in `detectron2` (both of my remarks are, actually)  with their [YACS configs](https://detectron2.readthedocs.io/en/latest/tutorials/configs.html). 

In short, in their YAML config, there's a top-level key `_BASE_`, which specifies the `base config` path.
You may also want to check [`load_yaml_with_base` function definition](https://detectron2.readthedocs.io/en/latest/_modules/fvcore/common/config.html#CfgNode.load_yaml_with_base). 

Here's [an example](https://github.com/facebookresearch/unbiased-teacher/blob/main/configs/coco_supervision/faster_rcnn_R_50_FPN_sup05_run1.yaml) -- `_BASE_` is defined on the first line.

With regard to the second remark, they provide scripts which can be used like this (see again [YACS configs](https://detectron2.readthedocs.io/en/latest/tutorials/configs.html)).
```
/demo.py --config-file config.yaml [--other-options] \
  --opts MODEL.WEIGHTS /path/to/weights INPUT.MIN_SIZE_TEST 1000
```

**Comment by samet-akcay** [comment]
@DominikFilipiak, we are already working on this, which will be available in v1.0.
